.pre-commit-config.yaml

```
@@ -57,6 +57,13 @@ repos:
 
   - repo: local
     hooks:
+      - id: check-version-references
+        name: Check for outdated version references in .tf files
+        language: system
+        entry: bash ci/check-version-references.sh
+        files: \.tf$
+        pass_filenames: false
+        stages: [commit, manual]
```

```suggestion
      - id: check-examples-versions
        name: Check for outdated version references in examples
        language: system
        entry: >-
          sh -c '
            latest_release=$(git tag --list --sort=-version:refname | grep -m1 -x -E "[0-9]+\.[0-9]+\.[0-9]+")

            # Only run on pre-push when pushing the latest release tag
            [ -z "${PRE_COMMIT_LOCAL_BRANCH}" ] || [ "${PRE_COMMIT_LOCAL_BRANCH}" != "refs/tags/${latest_release}" ] && exit 0

            git ls-files 'examples/*.tf' | while read -r f; do
              sed -i.precommit.bak -E -e "s_(\"git::https:\/\/github\.com\/DataDog\/terraform-module-datadog-agentless-scanner.*\?ref)=.*\"_\1=${latest_release}\"_" "$f"
              rm -f "${f}.precommit.bak"
            done
          '
        always_run: true
        stages: [pre-commit, pre-push]
```

Also add at the top of the file:
```yaml
default_stages: [pre-commit]
```

---

To install pre-push hooks:
```
pre-commit install --hook-type pre-push
```